include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1101 @@
+
+        auto v = ranges::iterator_range{begin_it, end_it} | seqan3::view::convert<value_type> | seqan3::view::to_rank;
+        data.insert(data.begin() + pos_as_num, ranges::begin(v), ranges::end(v));
+
+        return begin() + pos_as_num;
+    }
+
+    /*!\brief Inserts elements from initializer list before position in the container.
+     * \param   pos Iterator before which the content will be inserted. `pos` may be the end() iterator.
+     * \param ilist Initializer list with values to insert.
+     * \returns     Iterator pointing to the first element inserted, or pos if `ilist` is empty.
+     *
+     * Causes reallocation if the new size() is greater than the old capacity(). If the new size() is greater
+     * than capacity(), all iterators and references are invalidated. Otherwise, only the iterators and
+     * references before the insertion point remain valid. The past-the-end iterator is also invalidated.
+     *
+     * ### Complexity
+     *
+     * Worst-case linear in concat_size(). This is a drawback over e.g. `std::vector<std::vector<alphabet>>`.
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     */
+    iterator insert(const_iterator pos, std::initializer_list<value_type> const & ilist)
+    {
+        return insert(pos, ilist.begin(), ilist.end());
+    }
+
+    /*!\brief Removes specified elements from the container.
+     * \param begin_it Begin of range to erase.
+     * \param   end_it Behind the end of range to erase.
+     * \returns        Iterator following the last element removed. If the iterator pos refers to the last element,
+     *                 the end() iterator is returned.
+     *
+     * Invalidates iterators and references at or after the point of the erase, including the end() iterator.
+     *
+     * The iterator first does not need to be dereferenceable if first==end_it: erasing an empty range is a no-op.
+     *
+     * ### Complexity
+     *
+     * Linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     */
+    iterator erase(const_iterator begin_it, const_iterator end_it)
+    {
+        if (begin_it >= end_it) // [[unlikely]]
+            return begin() + std::distance(cbegin(), end_it);
+
+        auto const begin_it_pos = std::distance(cbegin(), begin_it);
+        auto const end_it_pos = std::distance(cbegin(), end_it);
+
+        data.erase(data.cbegin() + begin_it_pos,
+                   data.cbegin() + end_it_pos);
+
+        return begin() + begin_it_pos;
+    }
+
+    /*!\brief Removes specified elements from the container.
+     * \param   pos Remove the element at pos.
+     * \returns     Iterator following the last element removed. If the iterator pos refers to the last element,
```

Comment:
I always prefer emphasising variable names like `pos`. Can you check all places?